### ISSUE TITLE
usage of operator without internal user realm

### ISSUE THREAD

**Issue by rursprung** [Opster/opensearch-k8s-operator#221]
i admittedly haven't tried the operator yet, but based on the code it seems that it does everything with an admin user from the internal user realm (using basic auth with username + password).

are there any plans to support other modes? namely, we'd like to use the operator but, if possible, run without the internal user realm and instead only use JWT.

i currently see two options here (more might exist):
* the operator uses an admin certificate to manage the cluster (and can organise the certificate on its own; e.g. using #141)
* the operator provides a callback which can be implemented to get a valid auth token (which can be the basic auth header, a JWT header, a certificate, etc.) and then uses that

note: we currently don't use OpenSearch Dashboards, but i've also raised https://github.com/opensearch-project/OpenSearch-Dashboards/issues/970 for this a while ago.

**Comment by swoehrl-mw** [comment]
You are correct that currently the operator uses the internal admin user with basic auth for interacting with opensearch clusters.
There are currently no concrete plans to support anything else, but I can see the certificate option being viable. We already use an admin certificate to apply the securityconfig with securityadmin.sh so that would be the same direction.

**Comment by synhershko** [comment]
This is now supported: `spec.security.config.operatorClientCert` lets the operator authenticate to the cluster with an mTLS client cert instead of the internal admin user (added in #1414, documented in the user guide). That covers the admin-certificate option proposed here, so closing. If you still need the auth-callback/JWT variant, feel free to open a dedicated issue for it.
